apps/src/lab2/views/components/Instructions/ResourcePanel/index.tsx

```diff
@@ -313,9 +313,12 @@ const ResourcePanel: React.FC<ResourcePanelProps> = ({
   }, [currentTab, availableTabs]);
 
   useEffect(() => {
-    // Reset current tab to instructions when switching levels or viewAsUserId
+    // Reset current tab to instructions when switching levels or viewAsUserId.
     setCurrentTab(Tabs.Instructions);
-  }, [levelId, viewAsUserId]);
+    // Reset the permanent read-only state when the level changes.
+    isPermanentlyReadOnlyRef.current =
+      isReadOnly && !isRunning && !isValidating;
+  }, [levelId, viewAsUserId, isReadOnly, isRunning, isValidating]);
```

Comment:
With this change, if `isValidating`, `isRunning`, `isReadOnly` change we will switch to the instructions tab and set the readonly ref. If `isReadonly`, `isRunning`, or `isValidating` change, do we want the permanently readonly ref to change? I think that's why it was set at mount 🤔

Reply:
Yes - was able to add new selector - will update!